msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -959,6 +1119,43 @@ public void onCancel() {
         CommandDispatcher.submitSilent(command);
     }
 
+    private static NativeAuthPublicClientApplication createNativeAuthApplication(@NonNull final NativeAuthPublicClientApplicationConfiguration config,
+                                                    @Nullable final String clientId,
+                                                    @Nullable final String authority,
+                                                    @Nullable final String redirectUri,
+                                                    @Nullable final List<String> challengeTypes) throws BaseException {
+        if (clientId != null) {
+            config.setClientId(clientId);
+        }
+
+        if (authority != null) {
+            config.getAuthorities().clear();
+
+            final Authority authorityObject = Authority.getAuthorityFromAuthorityUrl(authority);
+            authorityObject.setDefault(true);
+            config.getAuthorities().add(authorityObject);
+        }
+
+        if (redirectUri != null) {
+            config.setRedirectUri(redirectUri);
+        }
+
+        if (challengeTypes != null) {
+            config.setChallengeTypes(challengeTypes);
+        }
+
+        // Check whether account mode is set to SINGLE
+        validateAccountModeConfiguration(config);
+
+        // TODO check with MSAL team
```

Comment:
This seems like an old TODO, what is it referring to?

Reply:
The comment was on if we should set the shared device flag to false. We had discussed in one of our previous chats and the result was that we should remove it. I have removed the TODO too.